src/Microsoft.OData.Client/InternalODataRequestMessage.cs

```diff
@@ -115,6 +115,15 @@ public override int Timeout
             set { throw new NotSupportedException(); }
         }
 
+        /// <summary>
+        /// Gets or sets the read and write timeout (in seconds) for this request.
+        /// </summary>
+        public override int ReadWriteTimeout

```

Comment:
What is this class for? It just throws exceptions...? Should it probably be removed?

Reply:
@g2mula, the new property is another timeout property of an <code> HttpWebRequest </code>. Basically, the <code> HttpWebRequest </code> that <code> Microsoft.OData.Client </code> uses has a Timeout property and a ReadWriteTimeout property. These properties have default values but you can update the default values if your request takes more than the default values to evaluate. Currently you can update the Timeout property but cannot update the ReadWriteTimeout property.

Reply:
Asking about this class, the InternalODataRequestMessage class, is it used anywhere?

Reply:
Yes there is a place where that class is required to throw exceptions if called.